server/src/metrics/client_impact_metrics.rs

```diff
@@ -13,11 +13,11 @@ pub struct ImpactMetricsKey {
     pub environment: String,
 }
 
-impl From<ImpactMetricEnv> for ImpactMetricsKey {
-    fn from(value: ImpactMetricEnv) -> Self {
+impl From<&ImpactMetricEnv> for ImpactMetricsKey {
```

Comment:
Smoother `From`. This dodges cloning the entire `ImpactMetric` and instead just clones what it needs